guides/pull-requests.md

```diff
@@ -0,0 +1,40 @@
+---
+explains: How to suggest changes and how to review them
+---
+
+# Pull requests
+
+Pull requests (as they are named on GitHub) or Merge requests (as they are named on GitLab) are proposed changes to a codebase. They are submitted by a contributor to a project and accepted or rejected by a repository's maintainers. Pull requests each have their own discussion forum enabling a collaborative review of the committed code.
+
+## Making a Pull Request
+
+When you make a Pull Request, make sure the code you are suggesting and tidy, as it will be reviewed by the project maintainers line-by-line. 
+
+Try to stay within the scope of what you are trying to fix. PRs with other fixes inside of them—even tiny corrections of typos—often make it hard to effectively review them. If you do want to fix a typo, make a new PR for that. It will surely be accepted and merged in no-time.
+
+Additionally, when making new commits to your branch, try to be clear about what you've done. Give the reviewer an idea of the changes you've introduced, and why. This will also make it easy for future contributors to understand your changes.
+
+For a basic guide on how PRs work read GitHub's excelent guide [Using Pull Requests](https://help.github.com/articles/using-pull-requests/).
```

Comment:
Totally agree with the previous remark about rebasing. It is a very important property for having a maintainable code base.
It can even further improved by using:
- Rebase when pulling by default:
git config --global pull.rebase true
- Enable rerere (reuse recorded resolution) globally to avoid solving the same
conflicts over and over
git config --global rerere.enabled true
- Auto stash pending changes when rebasing
This will make git stash away your changes when rebasing and attempt to apply the changes afterwards.
If the staged changes conflict with the new branch state you will get regular conflicts that you will need
to fix.
git config --global rebase.autoStash true
- And also very important:
Lock the master and develop branch against direct updates. All updates should go via pull requests

And a last remark: take git serious. Educating developers and providing good guidelines (global and project setups) are important.

Reply:
In order to improve the chance of the acceptance of a PR add tests! By providing test that show that your proposed change is working you do not only provide proof but also additional documentation. PR's should either include a statement why tests are not included (I already wonder how that statement would look like) or provide for tests that show the behavior of the software in both expected and in unexpected situations. Add code coverage to a project to check for minimal coverage requirements.

Reply:
I think mentioning the requirement for tests is definitely a good idea. Getting too far in to the weeds about Git is something I feel mixed about, on the one hand I think people should know their tools, on the other hand I hear the requirement for including more people in the process and making it easy.